### PR TITLE
added Istio section

### PR DESCRIPTION
**Pull request by phillipgibson**

### CODE REVIEW

File: workshops/advanced-aks/workshop.md [hunk 1]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+
+##### Configure MTLS Strict Mode for AKS Store Namespace
+
+Currently Istio configures workloads to use MTLS when calling other workloads, but the default permissive mode allows a service to accept traffic in plaintext or MTLS traffic. To ensure that the workloads we manage with the Istio add-on only accept MTLS communication, we will deploy a Peer Authentication policy to enforce only MTLS traffic for the workloads in the `aks-store` namespace.
+
+Prior to deploying the MTLS strict mode, let's verify that the `store-front` service will respond to a client not using MTLS. We will invoke a call from the `curl` pod to the `store-front` service and see if we get a response. Run the following command to curl to the `store-front` service.
+
+```bash
+kubectl exec -it $(kubectl get pod -l app=curl -o jsonpath="{.items[0].metadata.name}") -- curl store-front.aks-store.svc.cluster.local:80
+
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:366]: Is line 368 the expected output? If so, you should mention that.

File: workshops/advanced-aks/workshop.md [hunk 2]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+<!doctype html><html lang=""><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" href="/favicon.ico"><title>store-front</title><script defer="defer" src="/js/chunk-vendors.1541257f.js"></script><script defer="defer" src="/js/app.1a424918.js"></script><link href="/css/app.0f9f08e7.css" rel="stylesheet"></head><body><noscript><strong>We're sorry but store-front doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id="app"></div></body></html>
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:367]: Is line 367 the expected output? If so, you should mention that.

File: workshops/advanced-aks/workshop.md [hunk 3]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+```
+
+As you can see, HTML was returned to the client meaning that the `store-front` service successfully responed to the client in plaintext. Let's now apply the Peer Authentication policy that will enforce all services in the `aks-store` namespace to only use MTLS communication. Run the following command to configure the MTLS Peer Authentication policy.
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:370]: For clarity, remove the first sentence and add one before line 367 indicating that what follows is the expected output.

File: workshops/advanced-aks/workshop.md [hunk 4]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+kubectl apply -n aks-store -f - <<EOF
+apiVersion: security.istio.io/v1
+kind: PeerAuthentication
+metadata:
+  name: aks-store-default
+spec:
+  mtls:
+    mode: STRICT
+EOF
+
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:382]: For clarity, add a sentence indicating the expected output.

File: workshops/advanced-aks/workshop.md [hunk 5]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+peerauthentication.security.istio.io/aks-store-default created
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:383]: For clarity,  end the bash block at the end of the command (line 382) and start a new bash block preceded with a sentence indicating what follows is the command output.

File: workshops/advanced-aks/workshop.md [hunk 6]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+```
+
+Once the MTLS strict mode peer authentication policy has been applied, we will now see if we can again get a response back from the `store-front` service from a client not using MTLS. Run the following command to curl to the `store-front` service again.
+
+```bash
+kubectl exec -it $(kubectl get pod -l app=curl -o jsonpath="{.items[0].metadata.name}") -- curl store-front.aks-store.svc.cluster.local:80
+
+curl: (56) Recv failure: Connection reset by peer
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:391]: Same comment as line 382.

File: workshops/advanced-aks/workshop.md [hunk 7]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+By default, services are not accessible from outside the cluster. When managing your workloads with the Istio service mesh, you want to ensure that if you expose a service for communications outside the cluster, mesh policy configurations can still be preserved. 
+
+The AKS Istio AKS add-on comes with both a external and internal ingerss gateway that you can utilize to expose your services in the service mesh. In the following steps, we will show how to enable an external ingerss gateway to allow the `store-front` service to be reached from outside the cluster.
+
+The following command will enable the Istio ingress gateway on the AKS cluster. This may take several moments.
+
+```bash
+az aks mesh enable-ingress-gateway --resource-group myResourceGroup --name myAKSCluster --ingress-gateway-type external
+
+ / Running ..
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:408]: You can probably remove this.

File: workshops/advanced-aks/workshop.md [hunk 8]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+    "mode": "Istio"
+  },
+...
+```
+Use `kubectl get svc` to check the service mapped to the ingress gateway:
+
+```bash
+kubectl get svc aks-istio-ingressgateway-external -n aks-istio-ingress
+
+NAME                                TYPE           CLUSTER-IP     EXTERNAL-IP     PORT(S)
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:440]: Same as line 382

File: workshops/advanced-aks/workshop.md [hunk 9]
```diff
@@ -53,6 +53,455 @@ wt_id: WT.mc_id=containers-147656-pauyu
...
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: store-front.aks-store.svc.cluster.local
+        port:
+          number: 80
+EOF
+
+gateway.networking.istio.io/aks-store-gateway-external created
```
Comment (russd2357) [inline workshops/advanced-aks/workshop.md:487]: Same as line 382

**Review by russd2357 — COMMENTED** [review]
I left comments in places where I thought including the command output in the same bash block as the command would likely confuse the reader. For clarity it would be better to end the bash block with the command, add a sentence to the effect of "The output of the command looks like this:", then start a new bash block for the command output.

**Comment by phillipgibson** [comment]
> I left comments in places where I thought including the command output in the same bash block as the command would likely confuse the reader. For clarity it would be better to end the bash block with the command, add a sentence to the effect of "The output of the command looks like this:", then start a new bash block for the command output.

At this point I'm really just trying to get these sections into @pauldotyu branch and then we can clean it up on that branch before it's merged to main.

File: workshops/advanced-aks/workshop.md (hunk 3 above)
Comment (russd2357) [reply workshops/advanced-aks/workshop.md:370]: Will add an Issue as a reminder to clean this up later

**Pull request merged**